diff --git a/node_modules/expo-firebase-core/node_modules/expo-constants/android/build.gradle b/node_modules/expo-firebase-core/node_modules/expo-constants/android/build.gradle
index a813a80..37249a6 100644
--- a/node_modules/expo-firebase-core/node_modules/expo-constants/android/build.gradle
+++ b/node_modules/expo-firebase-core/node_modules/expo-constants/android/build.gradle
@@ -39,15 +39,20 @@ buildscript {
 
 // Creating sources with comments
 task androidSourcesJar(type: Jar) {
-  classifier = 'sources'
-  from android.sourceSets.main.java.srcDirs
+  archiveClassifier.set('sources')
+  if (project.hasProperty("android")) {
+    from android.sourceSets.main.java.srcDirs
+  }
 }
 
 afterEvaluate {
   publishing {
     publications {
       release(MavenPublication) {
-        from components.release
+        def releaseComponent = components.findByName("release")
+        if (releaseComponent != null) {
+          from releaseComponent
+        }
         // Add additional sourcesJar to artifacts
         artifact(androidSourcesJar)
       }
